src/java.naming/share/classes/com/sun/jndi/ldap/LdapPoolManager.java

```diff
@@ -395,47 +395,21 @@ private static void d(String msg, String o) {
         }
     }
 
-    private static final String getProperty(final String propName,
-        final String defVal) {
+    private static final String getProperty(final String propName, final String defVal) {
         return AccessController.doPrivileged(
-            new PrivilegedAction<String>() {
-            public String run() {
-                try {
-                    return System.getProperty(propName, defVal);
-                } catch (SecurityException e) {
-                    return defVal;
-                }
-            }
-        });
+                (PrivilegedAction<String>) () -> System.getProperty(propName, defVal)
+        );
     }
 
-    private static final int getInteger(final String propName,
-        final int defVal) {
-        Integer val = AccessController.doPrivileged(
-            new PrivilegedAction<Integer>() {
-            public Integer run() {
-                try {
-                    return Integer.getInteger(propName, defVal);
-                } catch (SecurityException e) {
-                    return defVal;
-                }
-            }
-        });
-        return val.intValue();
+    private static final int getInteger(final String propName, final int defVal) {
+        return AccessController.doPrivileged(
+                (PrivilegedAction<Integer>) () -> Integer.getInteger(propName, defVal).intValue()
+        );
     }
 
-    private static final long getLong(final String propName,
-        final long defVal) {
-        Long val = AccessController.doPrivileged(
-            new PrivilegedAction<Long>() {
-            public Long run() {
-                try {
-                    return Long.getLong(propName, defVal);
-                } catch (SecurityException e) {
-                    return defVal;
-                }
-            }
-        });
-        return val.longValue();
+    private static final long getLong(final String propName, final long defVal) {
+        return AccessController.doPrivileged(
+                (PrivilegedAction<Long>) () -> Long.getLong(propName, defVal).longValue()
+        );
```

Comment:
And GetIntegerAction here. Though it only supports an int value.

Reply:
Thanks for the suggestion Roger, I think the `privilegedGetProperty(prop, default)` for the `getProperty()` method looks great. 

WRT to using it for `getInt()` and `getLong()`, I think its reasonable to use other means for these methods in the interest of consistency due to, as you pointed out, only `int` being supported. Would you agree, or would it be better to use the same means in all 3 methods?

Reply:
Its a slippery slope that might require a bit more investigation to check the expected value sizes to see if a change to the number of bits in the value for each property might break something.  Technical debt can take you down a rabbit hole. Quickest to just convert to the lamba as you proposed.